src/huggingface_hub/_commit_api.py

```diff
@@ -503,7 +589,23 @@ def prepare_commit_payload(
                 "key": "deletedFolder" if operation.is_folder else "deletedFile",
                 "value": {"path": operation.path_in_repo},
             }
-        # 2.d. Never expected to happen
+        # 2.d. Case copying a file or folder
+        elif (
+            isinstance(operation, CommitOperationCopy)
+            and (operation.src_path_in_repo, operation.src_revision) in files_to_copy
```

Comment:
```suggestion
            and (operation.src_path_in_repo, operation.src_revision) in files_to_copy
```

this condition should now be guaranteed